pkg/component/crddeployer/crd.go

```diff
@@ -0,0 +1,155 @@
+// SPDX-FileCopyrightText: SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package crddeployer
+
+import (
+	"context"
+	"time"
+
+	"golang.org/x/exp/maps"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	"github.com/gardener/gardener/pkg/component"
+	"github.com/gardener/gardener/pkg/utils/flow"
+	kubernetesutils "github.com/gardener/gardener/pkg/utils/kubernetes"
+	"github.com/gardener/gardener/pkg/utils/kubernetes/health"
+	"github.com/gardener/gardener/pkg/utils/retry"
+)
+
+var (
+	// CRDWaitTimeout specifies the total time to wait for CRDs to become ready.
+	CRDWaitTimeout = 15 * time.Second
+)
+
+// crdDeployer is a DeployWaiter that can deploy CRDs and wait for them to be ready.
+type crdDeployer struct {
+	client            client.Client
+	applier           kubernetes.Applier
+	crdNameToManifest map[string]string
+}
+
+// NewCRDDeployer returns a DeployWaiter that can deploy CRDs and wait for them to be ready.
+func NewCRDDeployer(client client.Client, applier kubernetes.Applier, manifests []string) (component.DeployWaiter, error) {
```

Comment:
```suggestion
// New returns a new instance of DeployWaiter for CRDs.
func New(client client.Client, applier kubernetes.Applier, manifests []string) (component.DeployWaiter, error) {
```

The package is already called `crddeployer`, no need for repetition.